src/middleware.ts

```
@@ -0,0 +1,27 @@
+import { NextResponse } from 'next/server';
+import type { NextRequest } from 'next/server';
+import { getToken } from 'next-auth/jwt';
+
+export async function middleware(req: NextRequest) {
+  const token = await getToken({
+    req,
+    secret: process.env.AUTH_SECRET!,
```

I'd say let's throw an error outside of middleware if `AUTH_SECRET` or `PASSWORD_SALT` aren't set - would help avoiding runtime issues, we will catch them at build time